opentracing-util/src/main/java/io/opentracing/util/GlobalTracer.java

```diff
@@ -0,0 +1,116 @@
+
+    /**
+     * Singleton instance.
+     * <p>
+     * Since we cannot prevent people using {@linkplain #get() GlobalTracer.get()} as a constant,
+     * this guarantees that references obtained before, during or after initialization
+     * all behave as if obtained <em>after</em> initialization once properly initialized.<br>
+     * As a minor additional benefit it makes it harder to circumvent the {@link Tracer} API.
+     */
+    private static final GlobalTracer INSTANCE = new GlobalTracer();
+
+    /**
+     * The registered {@link Tracer} delegate or the {@link NoopTracer} if none was registered yet.
+     * Never {@code null}.
+     */
+    private static Tracer globalTracer = NoopTracerFactory.create();
+
+    private GlobalTracer() {
+    }
+
+    /**
+     * Returns the constant {@linkplain GlobalTracer}.
+     * <p>
+     * All methods are forwarded to the currently configured tracer.<br>
+     * Until a tracer is {@link #register(Tracer) explicitly configured},
+     * the {@link io.opentracing.NoopTracer NoopTracer} is used.
+     *
+     * @return The global tracer constant.
+     * @see #register(Tracer)
+     */
+    public static Tracer get() {
+        return INSTANCE;
+    }
+
+    /**
+     * Register a {@link Tracer} to back the behaviour of the {@link #get() global tracer}.
+     * <p>
+     * Registration is a one-time operation, attempting to call it more often will result in a runtime exception.
+     * <p>
+     * Every application intending to use the global tracer is responsible for registering it once
+     * during its initialization.
+     *
+     * @param tracer Tracer to use as global tracer.
+     * @throws RuntimeException if there is already a current tracer registered
+     */
+    public static synchronized void register(final Tracer tracer) {
+        if (tracer == null) {
+            throw new NullPointerException("Cannot register GlobalTracer <null>.");
+        }
+        if (tracer instanceof GlobalTracer) {
+            LOGGER.log(Level.FINE, "Attempted to register the GlobalTracer as delegate of itself.");
+            return; // no-op
+        }
+        if (globalTracer instanceof NoopTracer) {
+            globalTracer = tracer;
+        } else if (!globalTracer.equals(tracer)) { // be lenient for re-registration of same tracer
+            throw new IllegalStateException("There is already a current global Tracer registered.");
+        }
+    }
+
+    @Override
+    public SpanBuilder buildSpan(String operationName) {
+        return globalTracer.buildSpan(operationName);
+    }
+
+    @Override
+    public <C> void inject(SpanContext spanContext, Format<C> format, C carrier) {
+        globalTracer.inject(spanContext, format, carrier);
+    }
+
+    @Override
+    public <C> SpanContext extract(Format<C> format, C carrier) {
+        return globalTracer.extract(format, carrier);
+    }
+
+    @Override
+    public String toString() {
+        return GlobalTracer.class.getSimpleName();
```

Comment:
+1 both - good to know using global tracer, but also what tracer it is wrapping.